developer/src/kmcmplib/tests/gtest-compiler-test.cpp

```diff
@@ -401,6 +402,76 @@ TEST_F(CompilerTest, GetXStringImpl_type_single_quote_test) {
     // CERR_StringInVirtualKeySection *** TODO ***
 }
 
+// tests strings starting with 'i'
+TEST_F(CompilerTest, GetXStringImpl_type_i_test) {
+    KMX_WCHAR tstr[128];
+    fileKeyboard.version = VERSION_80;
+    KMX_WCHAR str[LINESIZE];
+    KMX_WCHAR output[GLOBAL_BUFSIZE];
+    PKMX_WCHAR newp = nullptr;
+    PFILE_STORE file_store = new FILE_STORE[100];
+    fileKeyboard.cxStoreArray = 3u;
+    fileKeyboard.dpStoreArray = file_store;
+    file_store[1].fIsStore = TRUE;
+    u16cpy(file_store[0].szName, u"a");
+    u16cpy(file_store[1].szName, u"b");
+    u16cpy(file_store[2].szName, u"c");
+
+    // index, CERR_InvalidInVirtualKeySection *** TODO ***
+
+    // index, no close delimiter => NULL
+    u16cpy(str, u"index(");
+    EXPECT_EQ(CERR_InvalidIndex, GetXStringImpl(tstr, &fileKeyboard, str, u"", output, 80, 0, &newp, FALSE));
+
+    // index, empty delimiters => empty string
+    u16cpy(str, u"index()");
+    EXPECT_EQ(CERR_InvalidIndex, GetXStringImpl(tstr, &fileKeyboard, str, u"", output, 80, 0, &newp, FALSE));
+
+    // index, no comma or space
+    u16cpy(str, u"index(b)");
+    EXPECT_EQ(CERR_InvalidIndex, GetXStringImpl(tstr, &fileKeyboard, str, u"", output, 80, 0, &newp, FALSE));
```

Comment:
Perhaps additional tests for whitespace inside parentheses?

Reply:
`index( )` (single space) also seems to suiffer from the runaway problem in `u16tok()`.

Reply:
`index( b)` and `index(b )` added